wdl-analysis/src/types/v1.rs

```diff
@@ -692,6 +693,33 @@ impl<'a, C: EvaluationContext> ExprTypeEvaluator<'a, C> {
         self.placeholders -= 1;
     }
 
+    fn invalid_placeholder_option(
+        ty: &Type,
+        _span: Span,
+        option: Option<PlaceholderOption>,
+    ) -> Diagnostic {
+        let message = match option {
+            Some(PlaceholderOption::Sep(_)) => format!(
+                "type mismatch for placeholder option `sep`: expected type `Array[P]` where P: any primitive type or Union, but found `{}`",
+                ty
+            ),
+            Some(PlaceholderOption::Default(_)) => format!(
+                "type mismatch for placeholder option `default`: expected an optional primitive type, but found `{}`",
+                ty
+            ),
+            Some(PlaceholderOption::TrueFalse(_)) => format!(
+                "type mismatch for placeholder option `truefalse`: expected type `Boolean`, but found `{}`",
+                ty
```

Comment:
For this one, we should probably use a `/` to delineate the placeholder name.
 
```suggestion
                "type mismatch for placeholder option `true/false`: expected type `Boolean`, but found `{ty}`"
```